tasties_app/models.py

```
@@ -46,6 +55,37 @@ class UnitChoices(models.TextChoices, Enum):
     )
     description = models.CharField(max_length=64, validators=[MinLengthValidator(1)])
 
+    def __str__(self):
+        return str(self.amount) + " " + self.measurement_unit + " " + self.description
+
+    @staticmethod
+    def is_enum_value(value):
+        """
+        Checks if the given value is a valid UnitChoices enum value.
+        """
+        for unit in Ingredient.UnitChoices:
+            if unit == value:
+                return True
+        return False
+
+    @staticmethod
+    def edit_info(ingredient, amount, measurement_unit, description):
+        """
+        Edits the ingredient's information
+            ingredient = ingredient to edit
+            amount = new amount value
+            measurement_unit = new measurement value
+            description = new description value
+        """
+        try:
+            ingredient.amount = amount
+            ingredient.measurement_unit = measurement_unit
+            ingredient.description = description
+            ingredient.full_clean()
+            ingredient.save()
+        except ValueError:
+            raise ValueError("Invalid ingredient info")
```

Why do you need to raise an exception here if you use the model validators?